plugins/chat/assets/javascripts/discourse/components/chat-live-pane.js

```diff
@@ -1269,12 +1268,14 @@ export default Component.extend({
     this.chatPreferredMode.setDrawer();
     this.appEvents.trigger("chat:open-channel", channel);
 
-    const previousRouteInfo = this.fullPageChat.exit();
-    if (previousRouteInfo) {
-      this._transitionToRoute(previousRouteInfo);
-    } else {
-      this.router.transitionTo(`discovery.${defaultHomepage()}`);
+    let previousURL = this.fullPageChat.exit();
+    if (!previousURL || previousURL === "/") {
+      previousURL = "discovery";
     }
+
+    this.router.replaceWith(previousURL).then(() => {
+      this.appEvents.trigger("chat:open-channel", channel);
```

Comment:
Out of curiosity, who listens to this event?

Reply:
topic-chat-float